src/PostComment.jsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { getUsers, postComment } from "./apis"
+
+
+export const PostComment = ({setListOfComments}) => {
+    const [isAddedToCommentList, setIsAddedToCommentList] = useState(false)
+    const [users, setUsers] = useState([])
+    const {review_id} = useParams()
+
+    useEffect(()=> {
+        getUsers().then((users) => {
+          const usernames = users.map((user) => {return user.username})
+            setUsers(usernames)
+        })
+    }, [])
+
+const handleSubmit = (e) => {
+    console.log(users)
+    e.preventDefault()
+    setIsAddedToCommentList(false)
+    const newComment = {
+        username: e.target[0].value,
+        //backend error so need author to display optismitic rendering of username:
+        author: e.target[0].value,
+        body: e.target[1].value,
+        votes: 0,
+        created_at: "Just now..."
+    }
+
+    if (users.includes(newComment.author)){
+        //optimistic rendering:
+        setListOfComments((comments) => {
+            const newComments = [...comments]
+            newComments.push(newComment)
+            return newComments
+        })
+        postComment(review_id, newComment).catch((err) => {
+
+            alert("sorry something went wrong, please try again later.")
+            setListOfComments((comments) => {
+                const newComments = [...comments]
+                    newComments.pop()
+                    return newComments
+                })
+        })
+
+        if (postComment){setIsAddedToCommentList(true)}
+    }
+    else{
+        alert("invalid username")
+    }
+    
+}
+
+if (isAddedToCommentList){
+    return (
+        <div>
+            <p>Comment Added!</p>
+            <button id="anotherComment" onClick={() => setIsAddedToCommentList(false)}>Post New Comment</button>
+            <br></br>
+        </div>
+    )
+}
+    return (
+        <div className="postComment">
```

Comment:
Is there a more appropiate semantic element we could use instead of a div? [https://www.w3schools.com/html/html5_semantic_elements.asp](url)